File: pics-legacy/src/main/java/com/picsauditing/report/fields/FilterType.java
package com.picsauditing.report.fields;


public enum FilterType {

	AccountID(QueryFilterOperator.Equals),
	Autocomplete(QueryFilterOperator.In),
	Boolean(QueryFilterOperator.Equals),
	Date(QueryFilterOperator.LessThan),
	Multiselect(QueryFilterOperator.In),
	Number(QueryFilterOperator.Equals),
	String(QueryFilterOperator.Contains),
	UserID(QueryFilterOperator.Equals);

	public QueryFilterOperator defaultOperator;

	FilterType(QueryFilterOperator defaultOperator) {
		this.defaultOperator = defaultOperator;
	}
}

File: pics-web/src/main/webapp/v7/js/extjs/pics/app/data/FilterType.js
Ext.define('PICS.data.FilterType', {
    statics: (function () {
        var types = {
            AccountID: 'AccountID',
            Autocomplete: 'Autocomplete',
            Boolean: 'Boolean',
            Date: 'Date',
            Multiselect: 'Multiselect',
            Number: 'Number',
            String: 'String',
            UserID: 'UserID'
        };
        
        return {
            getTypes: function () {
                return types;
            },
            
            AccountID: types.AccountID,
            Autocomplete: types.Autocomplete,
            Boolean: types.Boolean,
            Date: types.Date,
            Multiselect: types.Multiselect,
            Number: types.Number,
            String: types.String,
            UserID: types.UserID
        };
    }())
});